### PR TITLE
Mlego m65/m20 rp2040 keyboards

### PR DESCRIPTION
**Pull request by alinelena**
added two ortho support rp2040. all support basic matrix, oled, encoder, indicators and rgb.
one is 5x4 the other 5x13... box in lego.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mlego/m20/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/mlego/m20/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/mlego/m20/m20.c [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            if (clockwise) {
+                tap_code_delay(KC_VOLD, MEDIA_KEY_DELAY);
+            } else {
+                tap_code_delay(KC_VOLU, MEDIA_KEY_DELAY);
+            }
+#ifdef RGBLIGHT_ENABLE
+        }
+#endif
+    }
+}
```
Comment (drashna) [inline keyboards/mlego/m20/m20.c:125]: If anything, this should simply be replaced with encoder map support, at the keymap level.

File: keyboards/mlego/m20/m20.c [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    return OLED_ROTATION_180;
+}
+
+#    define SHOW_LOGO 5000
+bool oled_task_kb(void) {
+    if (!oled_task_user()) { return false; }
+    if ((timer_elapsed32(oled_logo_timer) < SHOW_LOGO)){
+        render_logo();
+    }else{
+      clear_screen();
```
Comment (drashna) [inline keyboards/mlego/m20/m20.c:199]: why not just call oled clear?

File: keyboards/mlego/m20/readme.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# mlego/m20
+
+![mlego/m20](https://imgur.com/B4jLOdn)
```
Comment (drashna) [inline keyboards/mlego/m20/readme.md:3]:
```suggestion
![mlego/m20](https://i.imgur.com/B4jLOdnh.jpeg)
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mlego/m20/keymaps/default/keymap.c (hunk 1 above)
Comment (alinelena) [reply keyboards/mlego/m20/keymaps/default/keymap.c]: done

File: keyboards/mlego/m20/m20.c (hunk 2 above)
Comment (alinelena) [reply keyboards/mlego/m20/m20.c:125]: no idea about the option, any reference?

File: keyboards/mlego/m20/m20.c (hunk 3 above)
Comment (alinelena) [reply keyboards/mlego/m20/m20.c:199]: is this some new feature? is the same code i use for mlego/m65

File: keyboards/mlego/m20/m20.c (hunk 2 above)
Comment (drashna) [reply keyboards/mlego/m20/m20.c:125]: It's new as of this last breaking changes cycle (0.17.0).  It uses an array like the keymap, but for the encoders, and is handled by "action_exec", so it is capable of using the full process_record stack, including custom keycodes (such as rgb keycodes, etc).

https://docs.qmk.fm/#/feature_encoders?id=encoder-map

There are a number of keymaps that are already using it, such as mine.  But the documents have examples for them. 

That said, it should be noted that it should only be enabled at the keymap level, and the configurator and via don't support it currently, but the underlying "dynamic keymaps" feature that via relies on does support it.

File: keyboards/mlego/m20/m20.c (hunk 3 above)
Comment (drashna) [reply keyboards/mlego/m20/m20.c:199]: Nope,`oled_clear()` has been in there for a long while.  But I've been growing more familiar with the oled code, as well.

File: keyboards/mlego/m20/m20.c (hunk 2 above)
Comment (alinelena) [reply keyboards/mlego/m20/m20.c]: thanks, used now and pushed.

File: keyboards/mlego/m20/m20.c (hunk 3 above)
Comment (alinelena) [reply keyboards/mlego/m20/m20.c:158]: ok now i remember, oled_clear() simply did not remove the logo image.
i even did a call to oled_render() after and no luck... I used that brutal method like was 1990 again

**Comment by alinelena** [comment]
this got closed via some messup of mine... I will reopen it clean

**Pull request closed without merging**